### ISSUE TITLE
Access is denied: maybe need a option to skip failed

### ISSUE THREAD

**Issue by fy0** [karrick/godirwalk#2]
```go
	err := godirwalk.Walk("D:/", &godirwalk.Options{
		Callback: func(osPathname string, de *godirwalk.Dirent) error {
			fmt.Printf("%s %s\n", de.ModeType(), osPathname)
			return nil
		},
	})
	fmt.Println(err)
```

```
d--------- D:\$RECYCLE.BIN\S-1-5-21-2045673143-4246043559-2053418641-500
cannot ReadDirents: cannot Open: open D:\$RECYCLE.BIN\S-1-5-21-2045673143-4246043559-2053418641-500: Access is denied.
```

**Comment by karrick** [comment]
That sounds like a super idea. I’ll do one up tomorrow.

**Comment by fy0** [comment]
Awesome!

**Comment by karrick** [comment]
I've added a few more examples but have not pushed them yet.

I'm also thinking about having `Walk` invoke `ErrorCallback` for errors returned by `Callback` and `PostChildrenCallback`. At first it did not make sense for when `Callback` returned an error, that it would invoke `ErrorCallback` to handle it. However, when making the examples, I realized that sometimes the `ErrorCallback` might simply be a log message followed by `SkipNode` return value. This allows the upstream code to use the same error handling without having to have as much boilerplate in the `Callback` and `PostChildrenCallback` functions.
